### PR TITLE
Enum switch

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: example/example_switch.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+constexpr std::string_view DoWork() {
+    return "default";
+}
+
+template <>
+constexpr std::string_view DoWork<Color::GREEN>() {
+    return "override";
+}
+
+struct SwitcherLambda1 {
```
Comment (Neargye) [inline example/example_switch.cpp]: Maybe use 
```
// helper type for the visitor #4
template<class... Ts> struct overloaded : Ts... { using Ts::operator()...; };
// explicit deduction guide (not needed as of C++20)
template<class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
```
For an example, this would be cleaner.

Comment (schaumb) [reply example/example_switch.cpp]: I replace it, thanks!

**Comment by Neargye** [comment]
On gcc7.5 not work https://godbolt.org/z/vf9K5xWds

**Comment by schaumb** [comment]
> On gcc7.5 not work https://godbolt.org/z/vf9K5xWds

with customizing enum names, it works
```
namespace magic_enum::customize {
    template<>
    constexpr std::string_view enum_name<Color>(Color c) noexcept {
        switch (c) {
            case Color::RED: return "RED";
            case Color::BLUE: return "BLUE";
            case Color::GREEN: return "GREEN";
        }
        return "";
    }
}
```

But this does not work because the code assumpts that enum has at least one element. 
I try to fix that. Probably need a test for this, with the latest compilers failing too on empty enums.

**Comment by schaumb** [comment]
@Neargye  I created a separate [PR](https://github.com/Neargye/magic_enum/pull/169) for the gcc7.5 issue

**Comment by Neargye** [comment]
It's ready to merge?

**Comment by schaumb** [comment]
> It's ready to merge?

@Neargye  there are some missing documentation part, but the code is functionally done

**Comment by schaumb** [comment]
Probably I am not the perfect person to write these documentation, my english is not that good.
